nibabel/cifti2/tests/test_cifti2.py

```diff
@@ -277,6 +277,30 @@ def test_matrix():
     assert_raises(TypeError, m.__setitem__, 0, ci.Cifti2Parcel())
     assert_raises(TypeError, m.insert, 0, ci.Cifti2Parcel())
 
+    mim_none = ci.Cifti2MatrixIndicesMap(None, 'CIFTI_INDEX_TYPE_LABELS')
```

Comment:
A set_index_map convenience function might make the logic here simpler for the user - the idea being that it ignores whatever is in the appliesToMatrixDimension list of the mapping passed in, and resets it to the dimension (or list of dimensions) that is the other argument, and resolves any conflicting dimension by first removing the dimension from existing mappings:

`myXML.set_index_map([0, 1], ci.Cifti2MatrixIndicesMap(None, 'CIFTI_INDEX_TYPE_LABELS')`

Which would only throw if the index map has no type or is otherwise semantically incomplete/invalid (i.e., voxel list but no volume space, contains a voxel outside the volume space).  It could also use an indices map constructor that doesn't require a list of dimensions to apply to.

While it does mean that a member of the input structure is ignored, the simpler usage and logic (overwrite existing) may be worth it.  As before, just a suggestion.